hotsos/core/plugins/kernel/kernlog/common.py

```diff
@@ -1,15 +1,47 @@
 import abc
 import os
 
+from searchkit.constraints import TimestampMatcherBase
+
+from hotsos.core.log import log
 from hotsos.core.config import HotSOSConfig
 from hotsos.core.host_helpers import CLIHelper, HostNetworkingHelper
-from hotsos.core.search import FileSearcher
+from hotsos.core.search import FileSearcher, SearchConstraintSearchSince
 
 KERNLOG_TS = r'\[\s*\d+\.\d+\]'
 KERNLOG_PREFIX = (r'(?:\S+\s+\d+\s+[\d:]+\s+\S+\s+\S+:\s+)?{}'.
                   format(KERNLOG_TS))
 
 
+class KernLogTimestampMatcher(TimestampMatcherBase):
+    """
+    kern.log has a slightly esoteric timetamp format so we have to do a little
+    juggling to get it into a standard format that can be used with search
+    constraints.
+    """
+
+    @property
+    def year(self):
+        """ Use current year as it is not normally included in the logs """
+        return CLIHelper().date(format='+%Y')
```

Comment:
This could be problematic towards the year ends (sosreport collected in Dec and hotsos ran in Jan) ;-)

Maybe, get the year from the sosreport itself, from `sos_commands/date/date` for example?

Reply:
if running against a sosreporet CLIHelper().date(format='+%Y') is going to return the year from sos_commands/date/date so should be doing what you ask already.